backend/layers/persistence/persistence.py

```diff
@@ -281,16 +281,10 @@ def get_dataset_versions_by_id(
                 canonical_ids.append(version.dataset_id)
                 artifact_ids.extend(version.artifacts)
 
-            if get_tombstoned:
-                canonical_dataset_query = session.query(DatasetTable).filter(DatasetTable.id.in_(canonical_ids))
-            else:
-                canonical_dataset_query = (
-                    session.query(DatasetTable)
-                    .filter(DatasetTable.id.in_(canonical_ids))
-                    .filter(DatasetTable.tombstone.is_(False))
-                )
+            canonical_dataset_query = session.query(DatasetTable).filter(DatasetTable.id.in_(canonical_ids))
+            if not get_tombstoned:
```

Comment:
nit: the vast majority of cases get_tombstoned will be False (maybe even every case in practice?) so I am not convinced we should break this into two queries

Reply:
it's not a new query, it's just adding an additional filter to the existing query

Reply:
ah got it